src/redis-cli.c

```diff
@@ -2497,14 +2533,34 @@ clusterManagerOptionDef clusterManagerOptions[] = {
 
 static void getRDB(clusterManagerNode *node);
 
-static void createClusterManagerCommand(char *cmdname, int argc, char **argv) {
+static int createClusterManagerCommand(char *cmdname, int argc, char **argv) {
     clusterManagerCommand *cmd = &config.cluster_manager_command;
     cmd->name = cmdname;
     cmd->argc = argc;
     cmd->argv = argc ? argv : NULL;
     if (isColorTerm()) cmd->flags |= CLUSTER_MANAGER_CMD_FLAG_COLOR;
-}
 
+    if (config.stdin_tag_arg) {
+        int i = 0, tag_match = 0;
+        cmd->stdin_arg = readArgFromStdin();
+
+        for (; i < argc; i++) {
+            if (strcasecmp(argv[i], config.stdin_tag_name)) continue;
+
+            tag_match = 1;
+            cmd->argv[i] = (char *)cmd->stdin_arg;
```

Comment:
don't we need to free the old value?

Reply:
i think no need? (char *)

Reply:
i didn't take a close look.. the question is where it comes from (if it is allocated on the heap or not).
you can also test is with valgrind maybe (might discover other leaks).

Reply:
The argv came from `int main(int argc, char **argv)`
I did tested it with valgrind, here what i found:  https://github.com/redis/redis/commit/c1c75225f4ba1e0e9194483d11ac3b412842859d

Reply:
ok. so you found other (pre-existing) leaks, but nothing about the new code in this PR, right?
and you also tested interactive mode?
anyway, you can fix these leaks as part of this PR, just mention that fact (unrelated fixes) in the top comment to be clear.

Reply:
> ok. so you found other (pre-existing) leaks, but nothing about the new code in this PR, right?

yes

> you also tested interactive mode?

yes, i also found others leaks

i will probably fix it in this PR (happy path: like non-interactive ping, --cluster call, interactive mode with ctrl C), will working in this weekend